hashin.py

```diff
@@ -112,6 +112,11 @@ def run_single_package(
     python_versions=None,
     verbose=False,
 ):
+    restriction = None
+    if ';' in spec:
+        spec, restriction = spec.split(';', 1)
+        spec = spec.strip()
+        restriction = restriction.strip()
```

Comment:
The extra lines for just stripping might be better done just once. 
`spec, restriction = [x.strip() for x in spec.split(';', 1)]`